### PR TITLE
Remove DisplayApp parameter from Screen base

### PR DESCRIPTION
**Pull request by Riksu9000**
DisplayApp was passed to all screens due to the Screen class requiring it, but it didn't use it for anything. Here I removed DisplayApp from the base class and removed all unused DisplayApp parameters.

### CODE REVIEW

File: src/displayapp/screens/settings/Settings.h [hunk 1]
```diff
@@ -20,6 +20,7 @@ namespace Pinetime {
         bool OnTouchEvent(Pinetime::Applications::TouchEvents event) override;
 
       private:
+        DisplayApp* app;
```
Comment (NeroBurner) [inline src/displayapp/screens/settings/Settings.h:23]: I think this reference is unused and can be removed, and later also the passed variable from the constructor can be removed

File: src/displayapp/screens/Notifications.h [hunk 2]
```diff
@@ -69,6 +69,7 @@ namespace Pinetime {
         };
 
       private:
+        DisplayApp* app;
```
Comment (NeroBurner) [inline src/displayapp/screens/Notifications.h:72]: unused, may be removed from class

**Review by NeroBurner — CHANGES_REQUESTED** [review]
great work slimming down the interfaces! I think some more classes/constructors may be slimmed down (cascading remove of unused `*app` variable)

File: src/displayapp/screens/Notifications.h (hunk 2 above)
Comment (Riksu9000) [reply src/displayapp/screens/Notifications.h:72]: This is used to call `SetFullRefresh()` function for the dismissal feature.

File: src/displayapp/screens/settings/Settings.h (hunk 1 above)
Comment (Riksu9000) [reply src/displayapp/screens/settings/Settings.h:23]: This is passed to the List class, for calling `app->StartApp()`

Comment (NeroBurner) [reply src/displayapp/screens/settings/Settings.h:23]: whoops, overlooked that

**Review by NeroBurner — APPROVED** [review]
seems like I was wrong, great work :)

**Review by JF002 — APPROVED** [review]
That's a huge PR! Good work! I guess Screen and other apps needed a ref to DisplayApp before, but previous refactoring have probably already fixed that!

**Pull request merged**